src/minio/core/policy_creator.py

```diff
@@ -0,0 +1,147 @@
+"""
+PolicyCreator module for creating new MinIO policy documents from scratch.
+
+This module focuses on creating policies with organized sections.
+For modifying existing policies, use PolicyBuilder.
+
+POLICY TYPES CREATED:
+
+1. USER HOME POLICIES (PolicyType.USER_HOME):
+   - Policy Name: "user-home-policy-{username}"
+   - Grants ADMIN access to user's personal warehouses:
+     * s3a://{bucket}/users-sql-warehouse/{username}/ (for Spark tables)
+     * s3a://{bucket}/users-general-warehouse/{username}/ (for general files)
+   - Includes full read/write/delete permissions for user's personal workspace
+   - Enables directory listing and bucket location access
+   - SHARING OPERATIONS: Only paths within user home policies can be shared/unshared
+     with other users or groups via the sharing API endpoints
+
+2. USER SYSTEM POLICIES (PolicyType.USER_SYSTEM):
+   - Policy Name: "user-system-policy-{username}"
+   - Grants access to system resources defined in SYSTEM_RESOURCE_CONFIG:
+     * Currently: s3a://cdm-spark-job-logs/spark-job-logs/{username}/ (WRITE access)
+     * Future: Additional system resources as configured
+   - Permission levels vary by resource (READ/WRITE/ADMIN)
+   - User-scoped resources include username in path, global resources don't
+
+3. GROUP POLICIES (PolicyType.GROUP_HOME):
+   - Policy Name: "group-policy-{groupname}"
+   - Grants WRITE access to group's shared workspace:
+     * s3a://{bucket}/groups-general-warehouse/{groupname}/
+   - Enables collaborative access for group members
+   - Members inherit group permissions through policy attachment
+"""
+
+import logging
+from typing import Dict, List
+
+from ..models.minio_config import MinIOConfig
+from ..models.policy import (
+    PolicyPermissionLevel,
+    PolicySectionType,
+    PolicyStatement,
+    PolicyType,
+)
+
+logger = logging.getLogger(__name__)
+
+# Global system resource configuration
+SYSTEM_RESOURCE_CONFIG = {
+    # Spark-logs-related resources
+    # Creates policy with access to s3a://cdm-spark-job-logs/spark-job-logs/username/
+    "spark": {
+        "bucket": "cdm-spark-job-logs",
+        "base_prefix": "spark-job-logs",
+        "user_scoped": True,  # Whether this resource is user-specific
+        "permission_level": PolicyPermissionLevel.WRITE,
+    },
+    # Future system resources can be added here:
+    #
+    # Creates policy with access to s3a://cdm-task-service/task-service/
+    # "task_service": {
+    #     "bucket": "cdm-task-service",
+    #     "base_prefix": "task-service",
+    #     "user_scoped": False,
+    #     "permission_level": PolicyPermissionLevel.READ,
+    # },
+    #
+    # Creates policy with access to s3a://cdm-user-scratch/
+    # "scratch": {
+    #     "bucket": "cdm-user-scratch",
+    #     "base_prefix": "",
+    #     "user_scoped": False,
+    #     "permission_level": PolicyPermissionLevel.WRITE,
+    # },
+}
+
+
+class PolicyCreator:
```

Comment:
My advice would be to integrate the policy creator and policy builder into a single policy manager that only emits valid, usable policies

Reply:
Ok. To me having to tell the policy builder not to do validity checks seems like a bad idea, but your codebase